ArduCopter/mode_auto.cpp

```diff
@@ -771,33 +771,99 @@ void Copter::ModeAuto::wp_run()
         set_throttle_takeoff();
         return;
     }
-
-    // process pilot's yaw input
     float target_yaw_rate = 0;
-    if (!copter.failsafe.radio) {
-        // get pilot's desired yaw rate
-        target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
-        if (!is_zero(target_yaw_rate)) {
-            auto_yaw.set_mode(AUTO_YAW_HOLD);
-        }
-    }
 
-    // set motors to full range
+    //Enable course locked add a2sandres
+    if (g.enbl_crs_lock == 1) {
+        if (!copter.failsafe.radio) {
+            // get pilot's desired yaw rate
+            target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
+            if (!is_zero(target_yaw_rate)) {
+                auto_yaw.set_mode(AUTO_YAW_LOOK_AT_NEXT_WP);
+            }
+        }
+    }
+    else {
+        if (!copter.failsafe.radio) {
+            // get pilot's desired yaw rate
+            target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
+            if (!is_zero(target_yaw_rate)) {
+                auto_yaw.set_mode(AUTO_YAW_HOLD);
+            }
+        }
+    }
+
+   // set motors to full range
     motors->set_desired_spool_state(AP_Motors::DESIRED_THROTTLE_UNLIMITED);
 
-    // run waypoint controller
-    copter.failsafe_terrain_set_status(wp_nav->update_wpnav());
+    // run waypoint controller 
+    copter.failsafe_terrain_set_status(wp_nav->update_wpnav());  
+      
+     // mode semi auto
+    if (g.auto_man_alt == 1) {
+        
+        // get pilot desired climb rate alt control Alt_Hold add a2sAndres
```

Comment:
Remove all `add a2sAndres` from comments.

Reply:
revised and changed